settlement/src/tests.rs

```diff
@@ -0,0 +1,181 @@
+                token_address: token_address.clone(),
+                amount: intial_balance,
+                receiver_address: temporary_receiver_address.clone(),
+            }),
+        },
+        "hi".to_owned(),
+        1234,
+    )
+    .unwrap();
+    csv.apply_commit(&Commit::Transaction(execute_tx.clone()))
+        .unwrap();
+    transactions.push(execute_tx.clone());
+    for i in 0..20 {
+        let tx = Transaction {
+            author: "doesn't matter".to_owned(),
+            timestamp: 0,
+            head: format!("commit {i}"),
+            body: "".to_owned(),
+            diff: Diff::None,
+        };
+        csv.apply_commit(&Commit::Transaction(tx.clone())).unwrap();
+        transactions.push(tx);
+    }
+
+    // Complete the block
+    let agenda = Agenda {
+        height: 1,
+        author: chain_info.reserved_state.consensus_leader_order[0].clone(),
+        timestamp: 0,
+        transactions_hash: Agenda::calculate_transactions_hash(&transactions),
+    };
+    csv.apply_commit(&Commit::Agenda(agenda.clone())).unwrap();
+    csv.apply_commit(&Commit::AgendaProof(AgendaProof {
+        height: 1,
+        agenda_hash: agenda.to_hash256(),
+        proof: chain_info
+            .validators
+            .iter()
+            .map(|private_key| TypedSignature::sign(&agenda, private_key).unwrap())
+            .collect::<Vec<_>>(),
+        timestamp: 0,
+    }))
+    .unwrap();
+    let block_header = BlockHeader {
+        author: chain_info.validators[0].public_key(),
+        prev_block_finalization_proof: chain_info.last_finalization_proof,
+        previous_hash: chain_info.last_finalized_header.to_hash256(),
+        height: 1,
+        timestamp: 0,
+        commit_merkle_root: BlockHeader::calculate_commit_merkle_root(
+            &csv.get_total_commits()[1..],
+        ),
+        repository_merkle_root: Hash256::zero(),
+        validator_set: chain_info.last_finalized_header.validator_set.clone(),
+        version: chain_info.last_finalized_header.version,
+    };
+    csv.apply_commit(&Commit::Block(block_header.clone()))
+        .unwrap();
+    let fp = chain_info
+        .validators
+        .iter()
+        .map(|private_key| TypedSignature::sign(&block_header, private_key).unwrap())
+        .collect::<Vec<_>>();
+    csv.verify_last_header_finalization(&fp).unwrap();
+
+    // Update light client
+    sc.update_treasury_light_client(block_header.clone(), fp)
+        .await
+        .unwrap();
+    assert_eq!(sc.get_light_client_header().await.unwrap(), block_header);
+
+    // Execute transfer
+    let commits = csv.get_total_commits();
+    let merkle_tree = OneshotMerkleTree::create(
+        commits[1..=(commits.len() - 2)]
+            .iter()
+            .map(|c| c.to_hash256())
+            .collect(),
+    );
+    let merkle_proof = merkle_tree
+        .create_merkle_proof(execute_tx.to_hash256())
+        .unwrap();
+    sc.execute(execute_tx, 1, merkle_proof).await.unwrap();
```

Comment:
The test probably fails if ```execute```or ```get_light_client_header``` is executed first while the tx from ```update_treasury_light_client``` is pending, doesn't it?

Reply:
You're right. I added sleep between them.